services/media/service/src/domains/common/utils/localization-utils.ts

```diff
@@ -0,0 +1,79 @@
+import { Dict } from '@axinom/mosaic-db-common';
+import {
+  DeleteLocalizationSourceEntityCommand,
+  LocalizationServiceMultiTenantMessagingSettings,
+  UpsertLocalizationSourceEntityCommand,
+  UpsertLocalizationSourceEntityFields,
+} from '@axinom/mosaic-messages';
+import { MosaicError } from '@axinom/mosaic-service-common';
+import { Table } from 'zapatos/schema';
+import { InternalErrors } from '../../../common';
+import { LocalizationMessageData } from '../models';
+
+export const getUpsertMessageData = (
+  serviceId: string,
+  entityType: string,
+  entityId: number,
+  fields: UpsertLocalizationSourceEntityFields,
+  entityTitle: string | undefined,
+  imageId: string | null | undefined,
+): LocalizationMessageData => {
+  const payload: UpsertLocalizationSourceEntityCommand = {
+    service_id: serviceId,
+    entity_type: entityType,
+    entity_id: entityId.toString(),
+    entity_title: entityTitle,
+    image_id: imageId,
+    fields,
+  };
+  return {
+    settings:
+      LocalizationServiceMultiTenantMessagingSettings.UpsertLocalizationSourceEntity,
+    payload,
+  };
+};
+
+export const getDeleteMessageData = (
+  serviceId: string,
+  entityType: string,
+  entityId: number,
+): LocalizationMessageData => {
+  const payload: DeleteLocalizationSourceEntityCommand = {
+    service_id: serviceId,
+    entity_type: entityType,
+    entity_id: entityId.toString(),
+  };
+  return {
+    settings:
+      LocalizationServiceMultiTenantMessagingSettings.DeleteLocalizationSourceEntity,
+    payload: payload,
+  };
+};
+
+export const assertLocalizationData: (
+  data: Dict<unknown> | undefined,
+  tableName: Table,
+) => asserts data is Dict<unknown> = (
+  data: Dict<unknown> | undefined,
+  tableName: Table,
+): asserts data is Dict<unknown> => {
+  if (!data) {
```

Comment:
To be really sure (as it is likely that it can come from a message that is not properly asserted): maybe add `typeof data !== 'object'`.

Maybe we can then rather generalize it as "assertObject" and further down "assertObjectField" or such?

Reply:
This data would come from logical replication service, so the responsibility
falls to the PgoutputPlugin to produce an object, which so far produced only null, undefined, or
object values (depending on operation type and db setup). Also, the thrown errors are localization-specific, so I did not
want to generalize it even more.

I think the current implementation should be good enough and would only require manual checks when
related npm packages are updated (e.g. PgOutputPlugin gets an update and somehow
starts producing data of non-object types), but if you think it would not hurt to check
for typeof 'object' anyway - then I would separate the check into a separate
`if` block and throw a more specific error in that case.